### ISSUE TITLE
umoci-0.4.1-0.1.fc29 fails to build: "TestMutateCache: unexpected layerDigest"

### ISSUE THREAD

**Issue by ganto** [ganto/copr-umoci#1]
The COPR build of `umoci-0.4.1-0.1.fc29` (using `golang-1.11-0.rc2.1.fc29`) fails the test suite at:
```
Executing(%check): /bin/sh -e /var/tmp/rpm-tmp.PC8IKD
+ umask 022
+ cd /builddir/build/BUILD
+ cd umoci-0.4.1
+ export GOPATH=/builddir/build/BUILDROOT/umoci-0.4.1-0.1.fc29.x86_64//usr/share/gocode:/usr/share/gocode
+ GOPATH=/builddir/build/BUILDROOT/umoci-0.4.1-0.1.fc29.x86_64//usr/share/gocode:/usr/share/gocode
+ go test -buildmode pie -compiler gc -ldflags ' -extldflags '\''-Wl,-z,relro   -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '\''' github.com/openSUSE/umoci/cmd/umoci
ok  	github.com/openSUSE/umoci/cmd/umoci	0.009s
+ go test -buildmode pie -compiler gc -ldflags ' -extldflags '\''-Wl,-z,relro   -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '\''' github.com/openSUSE/umoci/mutate
--- FAIL: TestMutateCache (0.00s)
    mutate_test.go:81: unexpected layerDigest: got sha256:96338a7c847bc582c82e4962a4285afcaf568e3913b0542b8745be27a418a806, expected sha256:9a98de6b2015d531559791e60518fd376ddc62d3062ee4f691b223c06175dbef
    mutate_test.go:103: unexpected configDigest: got sha256:ddcc2a93d5b0bcdcb571431c3607d84abe3752406f7c631a898340e6e7e61ed0, expected sha256:1d043a5807e0ca5bcde233f14a79928f9aa5ccecdd4a8e4bf4cdd0b557090f91
    mutate_test.go:130: unexpected manifestDigest: got sha256:0e8b342d2b01241b3f0197d0210ed5c0012d01817881defc1464e000f5b08f4d, expected sha256:8b8f01aa9da2942d5dde5032b501004957f060b027c27162346972cff9220200
    mutate_test.go:168: manifest.Config.Digest is not cached
    mutate_test.go:177: manifest.Layers.Digest is not cached
--- FAIL: TestMutateAdd (0.00s)
    mutate_test.go:81: unexpected layerDigest: got sha256:96338a7c847bc582c82e4962a4285afcaf568e3913b0542b8745be27a418a806, expected sha256:9a98de6b2015d531559791e60518fd376ddc62d3062ee4f691b223c06175dbef
    mutate_test.go:103: unexpected configDigest: got sha256:ddcc2a93d5b0bcdcb571431c3607d84abe3752406f7c631a898340e6e7e61ed0, expected sha256:1d043a5807e0ca5bcde233f14a79928f9aa5ccecdd4a8e4bf4cdd0b557090f91
    mutate_test.go:130: unexpected manifestDigest: got sha256:0e8b342d2b01241b3f0197d0210ed5c0012d01817881defc1464e000f5b08f4d, expected sha256:8b8f01aa9da2942d5dde5032b501004957f060b027c27162346972cff9220200
    mutate_test.go:248: manifest.Layers[0].Digest is not the same!
--- FAIL: TestMutateAddNonDistributable (0.00s)
    mutate_test.go:81: unexpected layerDigest: got sha256:96338a7c847bc582c82e4962a4285afcaf568e3913b0542b8745be27a418a806, expected sha256:9a98de6b2015d531559791e60518fd376ddc62d3062ee4f691b223c06175dbef
    mutate_test.go:103: unexpected configDigest: got sha256:ddcc2a93d5b0bcdcb571431c3607d84abe3752406f7c631a898340e6e7e61ed0, expected sha256:1d043a5807e0ca5bcde233f14a79928f9aa5ccecdd4a8e4bf4cdd0b557090f91
    mutate_test.go:130: unexpected manifestDigest: got sha256:0e8b342d2b01241b3f0197d0210ed5c0012d01817881defc1464e000f5b08f4d, expected sha256:8b8f01aa9da2942d5dde5032b501004957f060b027c27162346972cff9220200
    mutate_test.go:328: manifest.Layers[0].Digest is not the same!
--- FAIL: TestMutateSet (0.00s)
    mutate_test.go:81: unexpected layerDigest: got sha256:96338a7c847bc582c82e4962a4285afcaf568e3913b0542b8745be27a418a806, expected sha256:9a98de6b2015d531559791e60518fd376ddc62d3062ee4f691b223c06175dbef
    mutate_test.go:103: unexpected configDigest: got sha256:ddcc2a93d5b0bcdcb571431c3607d84abe3752406f7c631a898340e6e7e61ed0, expected sha256:1d043a5807e0ca5bcde233f14a79928f9aa5ccecdd4a8e4bf4cdd0b557090f91
    mutate_test.go:130: unexpected manifestDigest: got sha256:0e8b342d2b01241b3f0197d0210ed5c0012d01817881defc1464e000f5b08f4d, expected sha256:8b8f01aa9da2942d5dde5032b501004957f060b027c27162346972cff9220200
--- FAIL: TestMutatePath (0.12s)
    mutate_test.go:81: unexpected layerDigest: got sha256:96338a7c847bc582c82e4962a4285afcaf568e3913b0542b8745be27a418a806, expected sha256:9a98de6b2015d531559791e60518fd376ddc62d3062ee4f691b223c06175dbef
    mutate_test.go:103: unexpected configDigest: got sha256:ddcc2a93d5b0bcdcb571431c3607d84abe3752406f7c631a898340e6e7e61ed0, expected sha256:1d043a5807e0ca5bcde233f14a79928f9aa5ccecdd4a8e4bf4cdd0b557090f91
    mutate_test.go:130: unexpected manifestDigest: got sha256:0e8b342d2b01241b3f0197d0210ed5c0012d01817881defc1464e000f5b08f4d, expected sha256:8b8f01aa9da2942d5dde5032b501004957f060b027c27162346972cff9220200
FAIL
FAIL	github.com/openSUSE/umoci/mutate	0.137s
```
Fedora 27 and 28 builds are successful.

**Comment by ganto** [comment]
Upstream bug, see openSUSE/umoci#269

**Comment by cyphar** [comment]
I've released umoci `0.4.3` which you can use. It contains the upstream fix for the test failure.

**Comment by ganto** [comment]
Thanks a lot. :+1: 

I updated the spec file and now the RPM is available for Fedora 29 and Rawhide too.
